sample/sample_test.go

```diff
@@ -0,0 +1,184 @@
+package sample
+
+import (
+	"fmt"
+	"math"
+	"slices"
+	"testing"
+
+	"gonum.org/v1/gonum/floats"
+)
+
+func TestTemperature(t *testing.T) {
+	logits, err := Temperature(0.5).Apply([]float64{-3, -2, -1, 0, 1, 2, 4})
+	if err != nil {
+		t.Fatal(err)
+	}
+	want := []float64{-14, -12, -10, -8, -6, -4, 0}
+	if !floats.Equal(logits, want) {
+		t.Fatalf("got: %v, want: %v", logits, want)
```

Comment:
`t.Fatal` shouldn't be used unless the failure case invalidates the rest of the test